### DISCUSSION TITLE
Recording PDF references externally

### DISCUSSION THREAD

**Discussion by gnusupport** [ahrm/sioyek#369] [Ideas]
I would like to know if this Sioyek PDF reader can record externally PDF references similarly like XPDF can do?

XPDF has mechanism to assign key to specific command and then to report externally for example page number of quote of text in PDF on specific page number or search string.

I would need that feature in order to index PDFs easily. Indexing as full text is not what I mean. I mean finding specific title on PDF page, or specific PDF page and then marking some text and pressing a key which would then invoke a command with the PDF file name location as argument, some marked text and page number, something like that.

Does it already exist here?

**Comment by ahrm** [comment]
Yes, here is a blog post on this feature in version 1.4:
https://ahrm.github.io/jekyll/update/2022/07/05/implementing-a-screen-reader-for-sioyek.html .

This feature has improved significantly in newer commits (no binary release yet). For example instead of `%1` we use descriptive names like `%{file_name}`.
This repository contains some scripts that I have written for newer versions of sioyek:
https://github.com/ahrm/sioyek-python-extensions
